docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,376 @@
+Get(ctx context.Context, hash tmbytes.HexBytes) (*header.ExtendedHeader, error)
+// GetByHeight returns the header of the given height from the node's header store.
+// If the header of the given height is not yet available, the request will hang
+// until it becomes available in the node's header store.
+GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error)
+// GetRangeByHeight returns the given range [from:to) of ExtendedHeaders.
+GetRangeByHeight(ctx context.Context, from, to uint64) ([]*ExtendedHeader, error)
+// Subscribe creates long-living Subscription for validated ExtendedHeaders.
+// Multiple Subscriptions can be created.
+Subscribe() (Subscription, error)
+// SyncState returns the current state of the header Syncer. 
+SyncState() sync.State
+// SyncWait blocks until the header Syncer is synced to network head. 
+SyncWait(ctx context.Context) error
+// SyncHead provides the Syncer's view of the current network head.
+SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+}
+```
+
+##### Shares
+
+```go
+  type SharesModule interface {
+    // GetShare returns the Share from the given data Root at the given row/col
+    // coordinates.
+    GetShare(ctx context.Context, dah *Root, row, col int) (Share, error)
+    // GetSharesByNamespace returns all shares of the given nID from the given data
+    // Root.
+    GetSharesByNamespace(
+      ctx context.Context,
+      root *Root, 
+      nID namespace.ID, 
+    ) ([]Share, error)
+    // SharesAvailable subjectively validates if Shares committed to the given data
+    // Root are available on the network. 
+    SharesAvailable(ctx context.Context, root *Root) error
+    // ProbabilityOfAvailability calculates the probability of the data square
+    // being available based on the number of samples collected.
+    ProbabilityOfAvailability() float64
+  }
+```
+
+##### P2P
+
+```go
+  type P2PModule interface {
+    Info
+    Peers
+    PeerInfo
+    
+    Connect
+    Disconnect
+    Connectedness
+    NATStatus
+    
+    BlockAdd
+    BlockRm
+    BlockList
+    
+    MutualAdd
+    MutualRm
+    MutualList
+    
+    BandwidthStats
+    BandwidthPeer
+    BandwidthProtocol
+    
+    ResourceState
+    ResourceLimit
+    ResourceSetLimit
+    
+    PubSubPeers
+    PubSubScore
+    
+    BitSwapStats
+    BitSwapLedger
+    
+    DHTFindPeer
+  }
+```
+
+### NodeModule
+
+```go
+  type NodeModule interface {
+    Type
+    Version
+    
+    LogLevelList(ctx context.Context) (map[string]string, error)
+    LogLevelSet(ctx context.Context, name, level string) error
+    
+    Start(ctx context.Context) error
+    Stop(ctx context.Context) error
+
+    // NodeModule provides AuthModule 
+    AuthModule
+  }
+```
+
+#### AuthModule
+
+```go
+type AuthModule interface {
+    SetModuleAuth(ctx context.Context, module string) error  // TODO
+    UnsetModuleAuth(ctx context.Context, module string) error // TODO
```

Comment:
This API limits to only one user who has auth/dawg/privileges, meaning the API doesn't allow us to have multiple authenticated users: CLI, Optimint, Astria. Or they should have shared access token, which is a bad security practice. 

We should be able to generate multiple access tokens through the API and potentially with another permissions dimension. So that the node operator can grant access to various apps.

The minimalistic example of such an API is https://github.com/filecoin-project/lotus/blob/master/api/api_common.go. Find Auth there.